source/core/index.ts

```diff
@@ -862,7 +862,7 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 									result.catch(handler);
 								} else if (event === 'abort') {
 									// eslint-disable-next-line promise/prefer-await-to-then
-									result.then((request: unknown): void => {
+									void result.then((request: unknown): void => {
```

Comment:
Are you 100% sure `result` will never reject at this point? Maybe do an async-IIFE with a try/catch just to be sure?

Reply:
If `result` throws then the error is caught here:

https://github.com/sindresorhus/got/blob/545e6a239ffcc4f10599565c2bfbd512d3bbcd43/source/core/index.ts#L1516-L1517

Since in that case `fn` would be `this._createCacheableRequest` which returns `result` (the `request` event).

But JS may not see that it's caught **later** and may throw unhandled rejection. The fix is just to add `.catch(() => {})`.